proposal-vote-script/server/proposals.go

```diff
@@ -0,0 +1,91 @@
+package server
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"os/exec"
+
+	"github.com/vitwit/cosmos-utils/proposal-vote-script/config"
+)
+
+// Vote is to vote for the proposals which are in voting period
+// and pending for user vote
+func Vote(cfg *config.Config) error {
+	var isVoted bool
+
+	ops := HTTPOptions{
+		Endpoint: cfg.LCDEndpoint + "/cosmos/gov/v1beta1/proposals?proposal_status=2",
+		Method:   http.MethodGet,
+	}
+
+	resp, err := HitHTTPTarget(ops)
+	if err != nil {
+		log.Printf("Error while getting voting period proposals: %v", err)
+		return err
+	}
+
+	var p Proposals
+	err = json.Unmarshal(resp.Body, &p)
+	if err != nil {
+		log.Printf("Error while unmarshelling vote proposal response: %v", err)
+		return err
+	}
+
+	log.Printf("Length of voting proposals : %v", len(p.Proposals))
+
+	for _, proposal := range p.Proposals {
+		log.Printf("Voting period proposal ID : %v", proposal.ProposalID)
+		isVoted = false
+
+		ops = HTTPOptions{
+			Endpoint: cfg.LCDEndpoint + "/cosmos/gov/v1beta1/proposals/" + proposal.ProposalID + "/votes/" + cfg.AccountAddress,
+			Method:   http.MethodGet,
+		}
+
+		v, err := GetVoterStatus(ops)
+		if err != nil {
+			log.Printf("Error while getting votes of the proposal: %v", err)
+			return err
+		}
+
+		log.Printf("vote response.. %v", v)
+
+		if v.Vote.Option == "VOTE_OPTION_YES" {
```

Comment:
Why are we checking only for YES?

Reply:
Yeah, it can be any option. Actually script will vote as yes, so I have taken that Anil.
I will replace it.

Reply:
Script needs to vote as yes or any other options also Anil?

Reply:
You can just check for nil. If the vote is not nil, it's not voted